### ISSUE TITLE
Consider a struct-based approach

### ISSUE THREAD

**Issue by asinghvi17** [tecosaur/CheckDoc.jl#5]
This package looks really cool!  I was looking at it and thought that for Makie.jl use and others it may need user extensibility.  Have you considered a struct-based approach, where each check is a type, and some common API method is defined for that type?

Something like this:

```julia
abstract type AbstractDocstringCheck end

struct CheckNonEmpty <: AbstractDocstringCheck end

function check(::CheckNonEmpty, ::Symbol, ::Dict, raw::String, ::Any)::Union{ShortIssue, Vector{ShortIssue}}
    if all(isspace, raw)
        ShortIssue(:error, "The documentation string should not be empty"), true
    end
end
```
or something like this.  Then users can call `check(module; checks = AbstractDocstringCheck[...])` and define their own checks elsewhere!

**Comment by asinghvi17** [comment]
This kind of already exists since you can `push!` to the const array defined in the main file, I thought formalizing the interface might be nice though.

**Comment by tecosaur** [comment]
I'm onboard with this! (just time poor :sweat_smile:)

**Comment by tecosaur** [comment]
Implemented in 4d9cfa525ea88f0b7a86e8f0e23964fd7ab11105
